### ISSUE TITLE
Clarification regarding Behaviors triggering order design

### ISSUE THREAD

**Issue by RaphaelDDL** [marionettejs/backbone.marionette#2696]
Hello there,

The docs for [Behaviors](http://marionettejs.com/docs/v2.4.2/marionette.behaviors.html) don't explain thoroughly about the feature but what we can [concur](http://garethdavies.me/2014/03/26/marionette-behaviors.html) [from](http://benmccormick.org/2015/03/23/staying-dry-with-marionette-behaviors/) [some](http://spin.atomicobject.com/2014/09/11/marionette-behaviors-overview/) [posts](http://stackoverflow.com/questions/24047479/when-or-why-would-i-use-a-marionette-behavior) is that is a mix-in to be used for DOM/UI interactions. So far so good.

Could you clarify why triggerMethod triggers events on [views first then on behaviors](https://github.com/marionettejs/backbone.marionette/blob/d182dc3f09b13a2603c142029d45994e1455105c/src/view.js#L273) - shouldn't `_triggerEventOnBehaviors` be called first (run common code) then call your view-specific method (for 'plus' code)? 

I could understand the applications of it being called later, for e.g., Datepicker Behavior that calls `$( this.ui.whatever ).datepicker();` at onRender. But as people called multiple times a 'mixin' and the docs don't clarify on that point, I was expecting something more like a 'super call' shortcut.  Example code:

    var Behavior1 = Marionette.Behavior.extend( {
        onShow: function () {
            $( this.ui.something ).text( 'behavior 1' );
        }
    } );

    var BodyView = Marionette.ItemView.extend( {
        ui: {
            something: '#app-something'
        },
        behaviors: {
            behavior1: {
                behaviorClass: Behavior1
            }
        },
        onShow: function () {
            $( this.ui.something ).text( 'view text' );
        }
    } );

This would set `something`'s text to 'view text' then later change it to 'behavior 1'.

I would expect this to be reflected the same way as if we created a View instead of a Behavior and  BodyView extending from it (instead of Marionette), where onShow would translate into:

        onShow: function () {
            //the common goes first
            ViewPretendingToBeBehavior.prototype.onShow.apply(this, arguments);
            //our addition later
            $( this.ui.something ).text( 'view text' );
        }

Is it planned to make something of this fashion for Behaviors or it was never meant for these cases? (therefore we having to use backbone.mix or similar for the above 'behavior' if we are to isolate behaviors that should happen before)

I felt this wouldn't be a SO question as is more oriented to the reasoning behind the design. But beg your pardon if that was a wrong assumption.

Thanks for your time reading it.

Best Regards,
Raphael

**Comment by jasonLaster** [comment]
Hi, thanks for asking. Behaviors should be independent pieces of interaction. If you're looking to use views to override default behavior then behaviors are not the right utility in my opinion. A straight mixin will suffice.

**Comment by RaphaelDDL** [comment]
Thanks for the reply.

Yeah, I was expecting it to behave like a general mixin engine (so I wouldn't need to include yet another plugin - bb.mix), so I tried the examples I showed and also tried create a common templateHelpers and a serializeData as behaviors. But of course they failed as those two don't even trigger events which behavior can listen, which I discovered later.

So, to sum up the Behavior, it is used to make fine-tuning and invoke initializers (datepicker, input masks, etc) which are things that need to be set after the normal view code.

Thanks once more :)
